src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -198,14 +198,16 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
                 LocalDataStoreInstance.LocalDataStore = keyStore;
             }
 
-            services = new DefaultServiceProvider();
+            grainFactory = new GrainFactory();
+            IDictionary<Type,object> singletons = new Dictionary<Type, object> { { typeof(IGrainFactory), grainFactory } };
+            services = new DefaultServiceProvider(singletons);
```

Comment:
That makes sense but it's becoming a bit more than *a very basic* and hence my concern. I don't see why would anyone care if the `DefaultServiceProvider` was using an existing container internally (they are still interfacing with `IServiceProvider`).